presto-main/src/main/java/com/facebook/presto/sql/planner/plan/DistinctLimitNode.java

```diff
@@ -83,18 +85,16 @@ public boolean isPartial()
         return hashSymbol;
     }
 
+    @JsonProperty
     public List<Symbol> getDistinctSymbols()
     {
-        if (hashSymbol.isPresent()) {
-            return ImmutableList.copyOf(Iterables.filter(getOutputSymbols(), not(hashSymbol.get()::equals)));
-        }
-        return getOutputSymbols();
+        return distinctSymbols;
     }
 
     @Override
     public List<Symbol> getOutputSymbols()
     {
-        return source.getOutputSymbols();
+        return distinctSymbols;
```

Comment:
But it can't be source.getOutputSymbols(). Otherwise you might end up with something like:

```
- DistinctLimit[x] ::= [x, y]
    - Project ::= [x,y]
```

Which would be non-sensical. What's the value of `y` coming out of DistinctLimit?